src/functions/DidSiopMetadata.ts

```diff
@@ -1,12 +1,143 @@
 import { SIOP, SIOPErrors } from '../types';
-import { DiscoveryMetadataPayload, RPRegistrationMetadataPayload, SubjectIdentifierType } from '../types/SIOP.types';
+import {
+  DiscoveryMetadataPayload,
+  RPRegistrationMetadataPayload,
+  SubjectIdentifierType,
+  SupportedTypes,
+} from '../types/SIOP.types';
 
-export function assertValidMetadata(opMetadata: DiscoveryMetadataPayload, rpMetadata: RPRegistrationMetadataPayload) {
-  verifyCredentialsFormat(rpMetadata.credential_formats_supported, opMetadata.credential_formats_supported);
+const allDidMethodsSupported = [
```

Comment:
Hmzz, DID methods are the actual part after did:<method>. So not including did:. The SIOP spec has that wrong. So lets not repeat their mistake ;)

I am also not sure whether I like this approach to begin with. Basically the RP and OP just need to accept supported methods between one another. Whether the uniresolver we use supports it or not is not really relevant. That is up to the RP and OP itself. They can always add/use a resolver of their choice. So if there is a DID method called 'shinynew' that both claim to support, then we also can expect that both can resolve them. It is then not up to us to not allow it because it is not in our internal list

Reply:
Also see the remark down below about using this list

Reply:
Fixed!